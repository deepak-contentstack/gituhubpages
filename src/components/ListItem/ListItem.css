@import '../../../tokens/variables.scss';
@import '../../../tokens/tokens.css';
@import '../../static/styles/css/typography.css';
@import '../../static/styles/css/common.css';

.ListItem__Wrapper {
    &__Content {
        color: $color-brand-black-base;
    }

    .ListItem {
        align-items: baseline;
    }

    .ListItem__Icon {
        margin: 0.188rem 0.5rem 0 !important;
    }
    
    .ListItem__Content,
    .ListItem__HelpText {
        white-space: nowrap;
        overflow: hidden;
        text-overflow: ellipsis;
        max-width: 94%;
    }
    
    &:hover {
        background: $color-brand-white-darker;
        border-radius: 0.25rem;
        cursor: pointer;
        color: $color-brand-primary-base;

        .ListItem__Content, .ListItem__HelpText {
            color: $color-brand-primary-base;
        }

        .ListItem__Icon > path {
            fill: $color-brand-primary-base;
        }
    }
}