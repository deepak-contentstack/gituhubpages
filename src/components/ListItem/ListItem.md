# ListItem Component
- ListItem is a component that can be used to display a list item in the list of components.

## ListItem Props :-

- content: It is the headline of the list item. It can be a string or a React node element

- description: It is the description of the list item. It can be a string or a React node element

- onClick: It is the callback function for the click event of the list item

- style: The style that will be applied on the list item component

- className: It will be applied on the list item component

- icon: Icon to be displayed in the component content

- testId: This value can be passed to be used for testing