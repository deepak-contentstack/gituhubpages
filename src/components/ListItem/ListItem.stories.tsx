import React from 'react'

import ListItem from './ListItem'
import notes from './ListItem.md'

const iconArray: any = ['NewTab', 'InternalLink', 'Reset', 'ResetWhite'];

export default {
    title: 'Components / ListItem',
    component: ListItem,
    argTypes: {
        content: {
            control: { type: 'text' },
            description: 'Provide a content title for the list item',
            table: {
                type: {
                    summary: 'text'
                },
                defaultValue: {
                    summary: 'InternalLink'
                }
            }
        },
        description: {
            control: { type: 'text' },
            description: 'Provide a description for the list item',
            table: {
                type: {
                    summary: 'text'
                }
            }
        },
        icon: {
            control: { type: 'select', options: iconArray },
            description: 'Determine the type of button you want to use',
            table: {
                type: {
                    summary: 'select'
                },
                defaultValue: {
                    summary: 'InternalLink'
                }
            }
        },
        style: {
            control: { type: 'disabled' },
            description: 'Style to the list item',
            table: {
                type: {
                    summary: 'text'
                }
            }
        },
        className: {
            control: { type: 'disabled' },
            description: 'ClassName to the list item',
            table: {
                type: {
                    summary: 'text'
                }
            }
        },
        testId: {
            control: { type: 'disabled' },
            description: 'TestId to the list item',
            table: {
                type: {
                    summary: 'text'
                }
            }
        }
    },
    parameters: {
        componentSubtitle:
            'ListItem is a component that can be used to display a list item in the list of components.',
        notes: { notes },
    },
}

export const Default = (args) => {
    return (
        <>
            <ListItem
                content={args.content}
                description={args.description}
                onClick={() => {
                    alert("click on first list item component")
                }}
                icon={args.icon}
            />
        </>
    )
}

Default.args = {
    content: 'Title for list item',
    description: 'Description for the list item',
    icon: 'InternalLink',
};