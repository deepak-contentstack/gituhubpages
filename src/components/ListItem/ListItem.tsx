import './ListItem.css'

import React from 'react'

import cn from 'classnames'

// import HelpText from '../HelpText/HelpText'
// import Icon, { IconType } from '../Icon2/Icon'
import Paragraph from '../Typography/Paragraph/Paragraph'

export type ListItemProps = {
  content: string | React.ReactNode;
  description?: string | React.ReactNode;
  onClick?: () => void;
  style?: React.CSSProperties;
  className?: string;
  // icon?: IconType;
  testId?: string;
}

const ListItem = (props: ListItemProps) => {
  return (
    <div
      onClick={() => {
        props.onClick && props.onClick();
      }}
      className={cn('pl-10 pb-10 pt-10 pr-10 ListItem__Wrapper', {
        [props.className]: !!props.className
      })}
      style={props.style}
    >
      <div className="flex ListItem">
        <Paragraph
          tagName="p"
          className={'font-size-large ListItem__Content'}
          text={props.content as string}
        />
        {/* {
          props.icon &&
          <Icon
            className="ListItem__Icon"
            icon={props.icon}
            size="tiny"
          />
        } */}
      </div>
      {/* {
        props.description &&
        <HelpText
          className={'ml-0 mt-0 ListItem__HelpText'}
        >
          {props.description}
        </HelpText>
      } */}
    </div>
  );
};

export default ListItem;
