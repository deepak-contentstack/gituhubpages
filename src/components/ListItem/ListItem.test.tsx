import React from 'react'

import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'

import ListItem from './ListItem'

const defaultProps = {
    content: "list item content",
    description: "list item description",
    onClick: () => { alert("clicked!") },
    style: { color: "#000" },
    className: "custom-class",
    testId: "data-testId",
}

describe('List Item Component', () => {
    it('renders the component with content prop', () => {
        const output = mount(
            <ListItem
                testId={defaultProps.testId}
                content={defaultProps.content}
            />
        );
        expect(toJson(output)).toMatchSnapshot();
    });

    it('renders the component with description prop', () => {
        const output = mount(
            <ListItem
                testId={defaultProps.testId}
                content={defaultProps.content}
                description={defaultProps.description}
            />
        );
        expect(toJson(output)).toMatchSnapshot();
    });

    it('renders the component with onClick prop', () => {
        const output = mount(
            <ListItem
                testId={defaultProps.testId}
                content={defaultProps.content}
                description={defaultProps.description}
                onClick={defaultProps.onClick}
            />
        );
        expect(toJson(output)).toMatchSnapshot();
    });

    it('renders the component with style prop', () => {
        const output = mount(
            <ListItem
                testId={defaultProps.testId}
                content={defaultProps.content}
                description={defaultProps.description}
                onClick={defaultProps.onClick}
                style={defaultProps.style}
            />
        );
        expect(toJson(output)).toMatchSnapshot();
    });

    it('renders the component with className prop', () => {
        const output = mount(
            <ListItem
                testId={defaultProps.testId}
                content={defaultProps.content}
                description={defaultProps.description}
                onClick={defaultProps.onClick}
                style={defaultProps.style}
                className={defaultProps.className}
            />
        );
        expect(toJson(output)).toMatchSnapshot();
    });

    it('renders the component with icon prop', () => {
        const output = mount(
            <ListItem
                testId={defaultProps.testId}
                content={defaultProps.content}
                description={defaultProps.description}
                onClick={defaultProps.onClick}
                style={defaultProps.style}
                className={defaultProps.className}
            //icon={"NewTab"}
            />
        );
        expect(toJson(output)).toMatchSnapshot();
    });
})