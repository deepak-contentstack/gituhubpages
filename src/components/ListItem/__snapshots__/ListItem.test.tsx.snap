// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Item Component renders the component with className prop 1`] = `
<ListItem
  className="custom-class"
  content="list item content"
  description="list item description"
  onClick={[Function]}
  style={
    Object {
      "color": "#000",
    }
  }
  testId="data-testId"
>
  <div
    className="pl-10 pb-10 pt-10 pr-10 ListItem__Wrapper custom-class"
    onClick={[Function]}
    style={
      Object {
        "color": "#000",
      }
    }
  >
    <div
      className="flex ListItem"
    >
      <Paragraph
        className="font-size-large ListItem__Content"
        tagName="p"
        testId="cs-paragraph-tag"
        text="list item content"
      >
        <p
          className="font-size-large ListItem__Content"
          data-test-id="cs-paragraph-tag"
        >
          list item content
        </p>
      </Paragraph>
    </div>
  </div>
</ListItem>
`;

exports[`List Item Component renders the component with content prop 1`] = `
<ListItem
  content="list item content"
  testId="data-testId"
>
  <div
    className="pl-10 pb-10 pt-10 pr-10 ListItem__Wrapper"
    onClick={[Function]}
  >
    <div
      className="flex ListItem"
    >
      <Paragraph
        className="font-size-large ListItem__Content"
        tagName="p"
        testId="cs-paragraph-tag"
        text="list item content"
      >
        <p
          className="font-size-large ListItem__Content"
          data-test-id="cs-paragraph-tag"
        >
          list item content
        </p>
      </Paragraph>
    </div>
  </div>
</ListItem>
`;

exports[`List Item Component renders the component with description prop 1`] = `
<ListItem
  content="list item content"
  description="list item description"
  testId="data-testId"
>
  <div
    className="pl-10 pb-10 pt-10 pr-10 ListItem__Wrapper"
    onClick={[Function]}
  >
    <div
      className="flex ListItem"
    >
      <Paragraph
        className="font-size-large ListItem__Content"
        tagName="p"
        testId="cs-paragraph-tag"
        text="list item content"
      >
        <p
          className="font-size-large ListItem__Content"
          data-test-id="cs-paragraph-tag"
        >
          list item content
        </p>
      </Paragraph>
    </div>
  </div>
</ListItem>
`;

exports[`List Item Component renders the component with icon prop 1`] = `
<ListItem
  className="custom-class"
  content="list item content"
  description="list item description"
  onClick={[Function]}
  style={
    Object {
      "color": "#000",
    }
  }
  testId="data-testId"
>
  <div
    className="pl-10 pb-10 pt-10 pr-10 ListItem__Wrapper custom-class"
    onClick={[Function]}
    style={
      Object {
        "color": "#000",
      }
    }
  >
    <div
      className="flex ListItem"
    >
      <Paragraph
        className="font-size-large ListItem__Content"
        tagName="p"
        testId="cs-paragraph-tag"
        text="list item content"
      >
        <p
          className="font-size-large ListItem__Content"
          data-test-id="cs-paragraph-tag"
        >
          list item content
        </p>
      </Paragraph>
    </div>
  </div>
</ListItem>
`;

exports[`List Item Component renders the component with onClick prop 1`] = `
<ListItem
  content="list item content"
  description="list item description"
  onClick={[Function]}
  testId="data-testId"
>
  <div
    className="pl-10 pb-10 pt-10 pr-10 ListItem__Wrapper"
    onClick={[Function]}
  >
    <div
      className="flex ListItem"
    >
      <Paragraph
        className="font-size-large ListItem__Content"
        tagName="p"
        testId="cs-paragraph-tag"
        text="list item content"
      >
        <p
          className="font-size-large ListItem__Content"
          data-test-id="cs-paragraph-tag"
        >
          list item content
        </p>
      </Paragraph>
    </div>
  </div>
</ListItem>
`;

exports[`List Item Component renders the component with style prop 1`] = `
<ListItem
  content="list item content"
  description="list item description"
  onClick={[Function]}
  style={
    Object {
      "color": "#000",
    }
  }
  testId="data-testId"
>
  <div
    className="pl-10 pb-10 pt-10 pr-10 ListItem__Wrapper"
    onClick={[Function]}
    style={
      Object {
        "color": "#000",
      }
    }
  >
    <div
      className="flex ListItem"
    >
      <Paragraph
        className="font-size-large ListItem__Content"
        tagName="p"
        testId="cs-paragraph-tag"
        text="list item content"
      >
        <p
          className="font-size-large ListItem__Content"
          data-test-id="cs-paragraph-tag"
        >
          list item content
        </p>
      </Paragraph>
    </div>
  </div>
</ListItem>
`;
